Entity/Components/HealthComponent.gd

```diff
@@ -7,32 +7,30 @@ class_name HealthComponent
 ## Emit when health is changed
 signal on_health_changed(new_health: int)
 
+signal on_block_changed(new_block: int)
+
 ## The maximum health the entity can have; this is a hard limit
-@export var max_health: float = 100
+@export var max_health: int = 100
 ## The health the entity currently has [br]
 ## Between 0 and max_health
-var current_health: float = 100 # ? change this to max health, or just unset since it's init by the ready
+var current_health: int = 100 # ? change this to max health, or just unset since it's init by the ready
+
+
+var current_block: int = 0
 
 ## Initialize the health component, putting health to max health
 ## This only happens at game start for the player, the value is then tracked inside the [PlayerManager] singleton [br]
 ## For enemies, this is called when the scene is instantiated (or more accurately, when each enemy is instanciated) [br]
 func _ready() -> void:
 	_set_health(max_health)
+	PhaseManager.on_phase_changed.connect(reset_block)
```

Comment:
`PhaseManager.on_phase_changed` has two arguments, `current_phase` and `old_phase` but `reset_block` doesn't take any argument
To work, it should ignore the `old_phase` and use the `new_phase` to check when it's the turn of the player